corehq/form_processor/backends/sql/dbaccessors.py

```diff
@@ -628,13 +630,19 @@ def save_new_form(form):
             operation.form_id = form.form_id
 
         try:
-            with form.attachment_writer() as attachment_writer, \
-                    transaction.atomic(using=form.db, savepoint=False):
+            db_names = [form.db]
+            if has_tracked_vault_entries(form):
+                db_names.append(VAULT_DB_NAME_FOR_WRITE)
+            with form.attachment_writer() as attachment_writer:
+                with ExitStack() as stack:
+                    for db_name in db_names:
+                        stack.enter_context(transaction.atomic(db_name, savepoint=False))
```

Comment:
```suggestion
            with form.attachment_writer() as attachment_writer, ExitStack() as stack:
                for db_name in db_names:
                    stack.enter_context(transaction.atomic(db_name, savepoint=False))
```

Reply:
@snopoke is this better/correct in some way?

Reply:
it's just less indentation and a bit more concise